platform/api/azure/storage_mit.go

```diff
@@ -91,16 +90,19 @@ func (a *API) ListBlobs(storageaccount, storagekey, container string, params sto
 
 	bsc := sc.GetBlobService()
 
-	resp, err := bsc.ListBlobs(container, params)
+	resp, err := bsc.GetContainerReference(container).ListBlobs(params)
 	if err != nil {
 		return nil, fmt.Errorf("failed listing blobs for %q: %v", container, err)
 	}
 	var res []storage.Blob
+	cont := bsc.GetContainerReference(container)
```

Comment:
This could be moved above the line 93, where you could then call `resp, err := cont.ListBlobs(params)`.